### PR TITLE
Fix filtering on joined fields

### PR DESCRIPTION
**Pull request by tlrobinson**

### CODE REVIEW

File: frontend/test/metabase-lib/lib/queries/structured/Filter.unit.spec.js [hunk 1]
```diff
@@ -72,5 +72,21 @@ describe("Filter", () => {
         filter(["segment", 1]).setDimension(["field-id", ORDERS.TOTAL.id]),
       ).toEqual([null, ["field-id", ORDERS.TOTAL.id]]);
     });
+    fit("should set joined-field for new filter clause", () => {
```
Comment (paulrosenzweig) [inline frontend/test/metabase-lib/lib/queries/structured/Filter.unit.spec.js:75]: typo

**Review by paulrosenzweig — APPROVED** [review]
LGTM!

**Comment by mazameli** [comment]
Manually verified that this fixes filtering on joined columns. 👍

**Comment by dacort** [comment]
Should this PR be against `release-0.35.x` branch?

**Comment by dacort** [comment]
Verified both issues as well. 👍

**Review by dacort — APPROVED** [review]

**Pull request merged**